pkg/core/fetch.go

```diff
@@ -418,44 +420,85 @@ func executeFetch(ctx context.Context, pLog zerolog.Logger, plugin plugin.Plugin
 // * wildcard expansion
 // * verify no unknown resources
 // * verify no duplicate resources
-func normalizeResources(ctx context.Context, provider plugin.Plugin, resources []string) ([]string, diag.Diagnostics) {
+func normalizeResources(ctx context.Context, provider plugin.Plugin, resources, skip []string) ([]string, diag.Diagnostics) {
 	s, err := provider.Provider().GetProviderSchema(ctx, &cqproto.GetProviderSchemaRequest{})
 	if err != nil {
 		return nil, diag.FromError(err, diag.INTERNAL)
 	}
-	return doNormalizeResources(resources, s.ResourceTables)
+
+	return doNormalizeResources(resources, skip, s.ResourceTables)
+}
+
+// doNormalizeResources matches the given two resource lists to all provider resources and returns the requested resources (excluding skip resources) as another list.
+func doNormalizeResources(resources, skip []string, all map[string]*schema.Table) ([]string, diag.Diagnostics) {
+	useRes, diags := doGlobResources(resources, false, all)
+	skipRes, dd := doGlobResources(skip, true, all)
+	return funk.Subtract(useRes, skipRes).([]string), diags.Add(dd)
 }
 
-// doNormalizeResources returns a canonical list of resources given a list of requested and all known resources.
-// It replaces wildcard resource with all resources. Error is returned if:
+// doGlobResources returns a canonical list of resources given a list of requested and all known resources.
+// It replaces wildcard resource with all resources in non-wild mode. Error is returned if:
 //
 // * wildcard is present and other explicit resource is requested;
 // * one of explicitly requested resources is not present in all known;
 // * some resource is specified more than once (duplicate).
-func doNormalizeResources(requested []string, all map[string]*schema.Table) ([]string, diag.Diagnostics) {
-	if len(requested) == 1 && requested[0] == "*" {
+func doGlobResources(requested []string, allowWild bool, all map[string]*schema.Table) ([]string, diag.Diagnostics) {
```

Comment:
nit: I think this function has multiple jobs, its better to have small functions rather then one big one.

- one to handle "*" 
- one to handle skip
- one to open glob patterns

Reply:
skip is handled already in the same fn that's the point of normalize calling glob.
the glob pattern here is special and very basic so I don't think a new fn is needed just to wrap strings.HasSuffix and few exceptions.

Reply:
Updated